Libraries/CMSIS/Device/Maxim/MAX32657/Include/max32657.h

```diff
@@ -601,52 +615,70 @@ We may want to handle GET_IRQ better...
 #define MXC_I3C_S ((mxc_i2c_regs_t *)MXC_BASE_I3C_S)
 
 #if IS_SECURE_ENVIRONMENT
+#define MXC_BASE_I3C MXC_BASE_I3C_S
 #define MXC_I3C MXC_I3C_S
 #else
+#define MXC_BASE_I3C MXC_BASE_I3C_NS
 #define MXC_I3C MXC_I3C_NS
 #endif
 
 /******************************************************************************/
 /*                                                                        DMA */
 #define MXC_DMA_CHANNELS (4)
-#define MXC_DMA_INSTANCES (1)
-// ^ Note: We have 2 DMA instances in hardware, but they are secure vs non-secure
-// instances.  Therefore we treat the part as if there is only 1.
+#define MXC_DMA_INSTANCES (2)
 
 /* Non-secure Mapping */
 #define MXC_BASE_DMA0_NS ((uint32_t)0x40028000UL)
 #define MXC_DMA0_NS ((mxc_dma_regs_t *)MXC_BASE_DMA0_NS)
-/* DMA0 instance only for secure mode. */
 
 /* Secure Mapping */
 // TODO(ME30): Is there actuall a secure mapping for DMA0?
+//             -Yes, DMA0 can be accessed from secure mode. Realizing this, I think
+//                  we would still have to define two DMA instances.
+//                  DMA0 can only access the non-secure mappings of the peripherals,
+//                  but DMA0 can be accessed in both Non-secure and Secure code.
+//                  DMA1 can access both secure and non-secure addresses of the peripherals,
+//                  but DMA1 can Only be accessed in Secure code.
```

Comment:
DMA0 can be _accessed_ from secure mode but my understanding is that the controller cannot access secure memory/peripherals.  I think we should still preserve the ability to access it at the register-level, but for this first pass at the drivers I think we should revert back to the previous setup

Reply:
Yes, you are correct - as mentioned in the PR description above:

> Both DMA0 and DMA1 need to be defined, so we can't use the single `MXC_DMA` instance anymore.
> 
> 1. DMA0 has a secure and non-secure mapping - accessible for secure and non-secure builds. However, DMA0 can not access secure memory or secure peripherals.
> 2. DMA1 only has a secure mapping - accessible for secure builds. But DMA1 can access both secure or non-secure memory and peripherals.

The diction for all the security can get confusing. If we do a secure build, there needs to be an option to access both DMA0 and DMA1.